File: javisagent/backend/src/schemas/translate.py
from pydantic import BaseModel
from typing import Optional


class CloneResponse(BaseModel):
    voice_id: str
    name: str


class VoiceInfo(BaseModel):
    voice_id: str
    name: str
    preview_url: Optional[str] = None


class WSMessageType:
    TTS_REQUEST = "tts_request"
    MEETING_START = "meeting_start"
    MEETING_STOP = "meeting_stop"
    TTS_AUDIO = "tts_audio"
    TTS_DONE = "tts_done"
    SUBTITLE = "subtitle"
    STATUS = "status"
    ERROR = "error"

File: javisagent/frontend/src/types/translate.ts
export interface VoiceInfo {
  voice_id: string;
  name: string;
  preview_url?: string;
}

export interface CloneResponse {
  voice_id: string;
  name: string;
}

export type WSMessageType =
  | "tts_request"
  | "meeting_start"
  | "meeting_stop"
  | "tts_audio"
  | "tts_done"
  | "subtitle"
  | "error"
  | "status";

export interface WSMessage {
  type: WSMessageType;
  data: Record<string, unknown>;
}

export interface SubtitleEntry {
  type: "transcript" | "translation" | "user_transcript";
  text: string;
  is_final: boolean;
  lang: "en" | "zh";
  timestamp: number;
}

export interface CompanionStatus {
  running: boolean;
  voice_id: string | null;
  pid: number | null;
}
